using_restricted_array.rb

```diff
@@ -7,41 +7,94 @@
 # Calculates the length of the restricted array. All values are integers.
 # The restricted_array is terminated by 'nil' i.e. array[length] = nil
 def length(array)
-  puts "NOT IMPLEMENTED"
+  indexes = 0
+  until array[indexes] == nil
+    indexes += 1
+  end
+  indexes
 end
 
 # Prints each integer values in the array
 def print_array(array)
-  puts "NOT IMPLEMENTED"
+  new_array = []
```

Comment:
I'm unsure why you're creating new_array. It's not getting used to print the array values. You don't need it - it's an unused, unneeded array.